src/sync/app.cpp

```diff
@@ -183,5 +190,30 @@ void App::login_with_credentials(const AppCredentials& credentials,
     }, handler);
 }
 
+void App::log_out(std::function<void (Optional<AppError>)> completion_block) const {
+    std::string route = util::format("%1/session", m_auth_route);
+
+    auto handler = [completion_block, this](const Response& response) {
+        if (auto error = check_for_errors(response)) {
+            return completion_block(error);
+        }
+
+        this->current_user()->log_out();
```

Comment:
This may crash if someone calls `log_out` twice concurrently. Best check if the result of `current_user` is valid first.